src/SiteInformation/ContactAndLocation/EditCashCropModal.js

```diff
@@ -0,0 +1,217 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Dialog,
+  DialogTitle,
+  Grid,
+  DialogContent,
+  Select,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  DialogActions,
+  Button,
+  Snackbar,
+} from '@material-ui/core';
+import { useSelector, useDispatch } from 'react-redux';
+import MuiAlert from '@material-ui/lab/Alert';
+import { setEditCashCropModalOpen } from '../../Store/actions';
+import { callAzureFunction } from '../../utils/SharedFunctions';
+import { useAuth0 } from '../../Auth/react-auth0-spa';
+import PropTypes from 'prop-types';
+import IssueDialogue from '../../Comments/IssueDialogue';
+
+// Helper function
+function Alert(props) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
+const EditCashCropModal = ({ setSnackbarDataGlobal }) => {
+  const open = useSelector((state) => state.tableData.editCashCropModalOpen);
+  const editCashCropModalData = useSelector((state) => state.tableData.editCashCropModalData);
+  const [farmCode, setFarmCode] = useState();
+  const [cashCrop, setCashCrop] = useState();
+  const [showOtherMessage, setShowOtherMessage] = useState(false);
+  const [backedCashCrop, setBackedCashCrop] = useState();
+  const dispatch = useDispatch();
+  const { getTokenSilently, user } = useAuth0();
+  const [snackbarData, setSnackbarData] = useState({
+    open: false,
+    text: '',
+    severity: 'success',
+  });
+
+  useEffect(() => {
+    if (Object.keys(editCashCropModalData).length !== 0) {
+      setFarmCode(editCashCropModalData.code);
+      const cashCrop = editCashCropModalData.cash_crop;
+      setCashCrop(cashCrop);
+      setBackedCashCrop(cashCrop);
+    }
+  }, [editCashCropModalData]);
+
+  const handleEditCashCropModalClose = (wipeInfo) => {
+    if (wipeInfo) {
+      setCashCrop(backedCashCrop);
+    } else {
+      setBackedCashCrop(cashCrop);
+    }
+    setShowOtherMessage(false);
+    dispatch(setEditCashCropModalOpen(!open));
+  };
+
+  const handleSelectChange = (event) => {
+    setCashCrop(event.target.value);
+  };
+
+  const handleUpdateCashCrop = () => {
+    const data = {
+      values: {
+        next_cash_crop: cashCrop,
+      },
+      conditions: {
+        code: farmCode,
+      },
+    };
+
+    callAzureFunction(data, 'crowndb/farm_history', 'POST', getTokenSilently).then((res) => {
+      if (res.response && res.response.status === 201) {
+        if (cashCrop === 'Other') {
```

Comment:
will you update state.sharedSiteInfo.valuesEdited in redux after successfully updating so that the table refreshes and gets the new data?